deploy_to_s3.py

```diff
@@ -128,16 +128,19 @@ def upload_file_to_s3(path, filename):
             # upload, no redirect
             return True, False
 
-    with ThreadPoolExecutor(max_workers=5) as executor:
+    with ProcessPoolExecutor(max_workers=8) as executor:
+        uploads = []
         for (dirpath, dirnames, filenames) in os.walk(site_dir):
             print('Enter', dirpath)
             files_to_upload = [f for f in filenames if not f[0] == '.']
-            uploads = executor.map(upload_file_to_s3, [dirpath] * len(files_to_upload), files_to_upload)
-
-            for (did_upload, did_rediect) in uploads:
-                if did_upload:
-                    uploaded += 1
-                if did_rediect:
-                    redirected += 1
+            for filename in files_to_upload:
+                uploads.append(executor.submit(upload_file_to_s3, dirpath, filename))
+
+        for future in uploads:
+            did_upload, did_redirect = future.result()
```

Comment:
There is this `concurrent.futures.as_completed` which allows to iterate on an array of `Future` objects as they resolve (https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.as_completed). In this case I guess it should not change much though.